ntoskrnl/inbv/inbv.c

```diff
@@ -920,28 +1048,49 @@ DisplayBootBitmap(IN BOOLEAN TextMode)
 #endif
 
 #ifdef INBV_ROTBAR_IMPLEMENTED
-        /* Draw the progress bar bit */
-        if (Bar) InbvBitBlt(Bar, 0, 0);
+        if (Bar)
+        {
+            /* Save previous screen pixels to buffer */
+            InbvScreenToBufferBlt(Buffer, 0, 0, 22, 9, 24);
+            /* Draw the progress bar bit */
+            InbvBitBlt(Bar, 0, 0);
+            /* Store it in global buffer */
+            InbvScreenToBufferBlt(RotBarBuffer, 0, 0, 22, 9, 24);
+            /* Restore screen pixels */
+            InbvBufferToScreenBlt(Buffer, 0, 0, 22, 9, 24);
+        }
+
+        if (TempRotBarSelection != RB_PROGRESS_BAR)
+        {
+            /* Hide simple progress bar if not used */
+            ShowProgressBar = FALSE;
+        }
+
+        /* Do we have a system thread? */
+        if (!RotBarThreadActive)
+        {
+            /* We don't, initialize the progress bar */
+            InbvAcquireLock();
```

Comment:
Why do you need to acquire a lock here?

Reply:
Indeed, it doesn't need here, since thread is not started yet at this point.

But this still doesn't fix a delay.